e2e/cypress/integration/messaging/direct_messages_spec.js

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// Group: @messaging
+
+import * as TIMEOUTS from '../../fixtures/timeouts';
+
+import users from '../../fixtures/users.json';
+
+const userActive = users['user-1'];
+
+describe('Messaging', () => {
+    beforeEach(() => {
+        // # Create new user that will eventually be inactive
+        cy.apiAdminLogin();
+        cy.apiCreateUser().then((res) => {
+            cy.wrap(res.user).as('userInactive');
+        });
+    });
+
+    it('M23349 - Deactivated users are not shown in Direct Messages modal', () => {
+        // # Deactivate inactive user
+        cy.get('@userInactive').then((userInactive) => {
+            // # Login as active user
+            cy.apiLogin(userActive);
+
+            // # Deactivate inactive user
+            cy.apiActivateUser(userInactive.id, false);
+
+            // # Visit main channel
+            cy.visit('/ad-1/channels/town-square');
```

Comment:
We're now not going to use "ad-1" for testing.  You may use the team generated from `cy.apiInitSetup()`.